StreamingAnalytics/LowLatencySample/Temperature-Epl/monitors/TemperatureEventMonitor.mon

```diff
@@ -0,0 +1,29 @@
+package lowlatency;
+using com.apama.aggregates.mean;
+/** If average temperature in 5 minutes is greater than 80 C, send a shutdown alert  */
+
+event Temperature {
+    float temperature;
+}
+
+event ShutDown {
+    string shutdownMessage;
+}
+
+monitor TempEventMonitor {
+    constant float FIVE_MINS_INTERVAL := 300.0;
+    constant float THRESHOLD_TEMPERATURE := 80.0;
+    action onload() {
+       monitor.subscribe("mqtt:sensors/temperature");
+       monitor.subscribe("mqtt:alerts/temperature");
+       stream<Temperature> temperatures := all Temperature();
+       from temp in temperatures within FIVE_MINS_INTERVAL
+         select Temperature(mean(temp.temperature)) as temperature{
+           if temperature.temperature > THRESHOLD_TEMPERATURE {
+                log temperature.temperature.toString() at INFO;
+                send ShutDown("ShutDown ! Temp is high") to "mqtt:alerts/temperature";
```

Comment:
I think this needs to be sent to the device, not to the cloud. Not sure if there is a convention for what the topic name should be.

Reply:
Yes , there is this topic name "alerts/temperature" where we need to send shutdown alert.